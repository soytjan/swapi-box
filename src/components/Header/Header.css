@import url("https://fonts.googleapis.com/css?family=Bungee+Hairline|Bungee+Outline|Roboto|Roboto+Condensed");
.header {
  height: 150px;
  background-image: url("./starrynight1.jpg");
  background-color: turquoise;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 1.8em; }
  .header h1 {
    color: white;
    font-family: "Bungee Outline", cursive;
    margin: 0;
    text-shadow: 3px 3px 4px black; }
  .header p {
    font-size: 16px; }

.fav-btn {
  background-color: #5F8773; }

.fav-box {
  display: flex;
  height: 40px;
  align-items: center;
  width: 170px;
  justify-content: space-between;
  background-color: #2B2B2D;
  color: white;
  padding: 0 10px;
  position: absolute;
  right: 50px; }
