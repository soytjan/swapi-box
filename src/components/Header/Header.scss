@import "../../variables.scss";

.header {
  height: 150px;
  background-image: url('./starrynight1.jpg');
  background-color: turquoise;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 2em;
  

  h1 {
    color: white;
    font-family: $header-font;
  }
}