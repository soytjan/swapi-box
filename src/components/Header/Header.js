import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import './Header.css';

const Header = ({favCount}) => {
  return (
    <div className='header'>
      <h1>SWAPI box</h1>
      <div className='fav-box'>
        <button className='fav-btn'><NavLink to='/favorites'>Favorites</NavLink></button>
        <p>{favCount}</p>
      </div>
    </div>
  )
}

Header.propTypes = {
  favCount: PropTypes.number,
};

export default Header;