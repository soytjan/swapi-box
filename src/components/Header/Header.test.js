import React from 'react';
import Header from './Header';
import { shallow } from 'enzyme';

describe('Header', () => {
  it('expect to exist', () => {
    expect(Header).toBeDefined();
  })

  it('matches snapshot', () => {
    const mockFavArray = [{ model: "Digger Crawler", name: "Sand Crawler", passengers: "30",vehicleClass:"wheeled" }];
    const renderedComponent = shallow(<Header favCount={mockFavArray.length} />)

    expect(renderedComponent).toMatchSnapshot();
  })

})