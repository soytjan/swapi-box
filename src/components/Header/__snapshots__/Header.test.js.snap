// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    favCount={1}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          SWAPI box
</h1>,
        <div
          className="fav-box"
>
          <button
                    className="fav-btn"
          >
                    <NavLink
                              activeClassName="active"
                              ariaCurrent="true"
                              to="/favorites"
                    >
                              Favorites
                    </NavLink>
          </button>
          <p>
                    1
          </p>
</div>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "SWAPI box",
        },
        "ref": null,
        "rendered": "SWAPI box",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="fav-btn"
>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            to="/favorites"
              >
                            Favorites
              </NavLink>
</button>,
            <p>
              1
</p>,
          ],
          "className": "fav-box",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                ariaCurrent="true"
                to="/favorites"
>
                Favorites
</NavLink>,
              "className": "fav-btn",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Favorites",
                "to": "/favorites",
              },
              "ref": null,
              "rendered": "Favorites",
              "type": [Function],
            },
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 1,
            },
            "ref": null,
            "rendered": 1,
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            SWAPI box
</h1>,
          <div
            className="fav-box"
>
            <button
                        className="fav-btn"
            >
                        <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    to="/favorites"
                        >
                                    Favorites
                        </NavLink>
            </button>
            <p>
                        1
            </p>
</div>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "SWAPI box",
          },
          "ref": null,
          "rendered": "SWAPI box",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="fav-btn"
>
                <NavLink
                                activeClassName="active"
                                ariaCurrent="true"
                                to="/favorites"
                >
                                Favorites
                </NavLink>
</button>,
              <p>
                1
</p>,
            ],
            "className": "fav-box",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  to="/favorites"
>
                  Favorites
</NavLink>,
                "className": "fav-btn",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": "Favorites",
                  "to": "/favorites",
                },
                "ref": null,
                "rendered": "Favorites",
                "type": [Function],
              },
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 1,
              },
              "ref": null,
              "rendered": 1,
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
